cpp/open3d/visualization/rendering/filament/GaussianSplatBuffers.cpp

```diff
@@ -0,0 +1,187 @@
+                                 .attribute(VertexAttribute::CUSTOM0, 3,
+                                            VertexBuffer::AttributeType::FLOAT4)
+                                 .attribute(VertexAttribute::CUSTOM1, 4,
+                                    VertexBuffer::AttributeType::FLOAT4)
+                                 .attribute(VertexAttribute::CUSTOM2, 5,
+                                    VertexBuffer::AttributeType::FLOAT4)
+                                 .attribute(VertexAttribute::CUSTOM3, 6,
+                                    VertexBuffer::AttributeType::FLOAT4)
+                                 .attribute(VertexAttribute::CUSTOM4, 7,
+                                    VertexBuffer::AttributeType::FLOAT4)
+                                 .attribute(VertexAttribute::CUSTOM5, 8,
+                                    VertexBuffer::AttributeType::FLOAT4)
+                                 .attribute(VertexAttribute::CUSTOM6, 9,
+                                    VertexBuffer::AttributeType::FLOAT4)
+                                 .build(engine);
+
+    VertexBufferHandle vb_handle;
+    if (vbuf) {
+        vb_handle = resource_mgr.AddVertexBuffer(vbuf);
+    } else {
+        return {};
+    }
+
+    const size_t vertex_array_size = n_vertices * 3 * sizeof(float);
+    float* vertex_array = static_cast<float*>(malloc(vertex_array_size));
+    memcpy(vertex_array, points.GetDataPtr(), vertex_array_size);
+    VertexBuffer::BufferDescriptor pts_descriptor(
+            vertex_array, vertex_array_size,
+            GeometryBuffersBuilder::DeallocateBuffer);
+    vbuf->setBufferAt(engine, 0, std::move(pts_descriptor));
+
+    const size_t color_array_size = n_vertices * 4 * sizeof(float);
+    float* color_array = static_cast<float*>(malloc(color_array_size));
+    if (geometry_.HasPointAttr("f_dc") && geometry_.HasPointAttr("opacity")) {
+        float* f_dc_ptr = static_cast<float*>(geometry_.GetPointAttr("f_dc").GetDataPtr());
+        float* opacity_ptr = static_cast<float*>(geometry_.GetPointAttr("opacity").GetDataPtr());
+        for (size_t i = 0; i < n_vertices; i++) {
+            std::memcpy(color_array + i * 4, f_dc_ptr + i * 3, 3 * sizeof(float));
+            std::memcpy(color_array + i * 4 + 3, opacity_ptr + i, sizeof(float));
+        }
+    } else {
+        for (size_t i = 0; i < n_vertices * 4; i++) {
+            color_array[i] = 1.f;
+        }
+    }
+    VertexBuffer::BufferDescriptor color_descriptor(
+                color_array, color_array_size,
+                GeometryBuffersBuilder::DeallocateBuffer);
+    vbuf->setBufferAt(engine, 1, std::move(color_descriptor));
+
+    const size_t rot_array_size = n_vertices * 4 * sizeof(float);
+    float* rot_array = static_cast<float*>(malloc(rot_array_size));
+    if (geometry_.HasPointAttr("rot")) {
+        std::memcpy(rot_array, geometry_.GetPointAttr("rot").GetDataPtr(),
+               rot_array_size);
+    } else {
```

Comment:
This if-else should be unnecessary, right? We are guaranteed that the mandatory attributes are present (f_dc, opacity, scale and rot).